src/controllers/molecules/RoomHeader.js

```diff
@@ -16,6 +16,72 @@ limitations under the License.
 
 'use strict';
 
+/*
+ * State vars:
+ * this.state.callState = OUTBOUND|INBOUND|IN_CALL|NO_CALL
+ */
+
+var dis = require("../../dispatcher");
+var CallHandler = require("../../CallHandler");
+
 module.exports = {
+
+    componentDidMount: function() {
+        this.dispatcherRef = dis.register(this.onAction);
+        if (this.props.room) {
+            var call = CallHandler.getCall(this.props.room.roomId);
+            var callState = call ? call.call_state : "ended";
+            this.setState({
+                call_state: callState
+            });
+        }
+    },
+
+    componentWillUnmount: function() {
+        dis.unregister(this.dispatcherRef);
+    },
+
+    onAction: function(payload) {
+        // if we were given a room_id to track, don't handle anything else.
+        if (payload.room_id && this.props.room && 
+                this.props.room.roomId !== payload.room_id) {
+            return;
+        }
+        if (payload.action !== 'call_state') {
+            return;
+        }
+        var call = CallHandler.getCall(payload.room_id);
+        var callState = call ? call.call_state : "ended";
+        this.setState({
+            call_state: callState
+        });
```

Comment:
Ah wait, I see: we get the call state from the horse's mouth anyway. This is good. Perhaps we can not bother putting the state in the payload?